core/templates/pages/exploration-editor-page/editor-navigation/editor-navbar-breadcrumb.component.spec.ts

```diff
@@ -16,14 +16,22 @@
  * @fileoverview Unit tests for editorNavbarBreadcrumb.
  */
 
+// TODO(#7222): Remove usage of UpgradedServices once upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+
 describe('Editor Navbar Breadcrumb directive', function() {
   var $rootScope = null;
   var $scope = null;
   var ExplorationTitleService = null;
   var FocusManagerService = null;
   var RouterService = null;
 
-  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
```

Comment:
Hi @brianrodri, is it possible to use TestBed here and below (see [here](https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide#when-should-the-upgraded-services-be-imported-in-the-test-file))?

Reply:
Hmm after taking a look it's not clear what needs to be done when a mix of AngularJS and Angular 8 imports are required. Should I use 2 separate beforeEach cases?

Reply:
I think you can do an `angular.mock.inject(..)` for AngularJS modules and then mock the required methods. Here's an example where we use an AngularJS service (StoryEditorStateService) along with migrated services:

https://github.com/oppia/oppia/blob/fa8d8296ef856ddd7dabd092091481e62c065c78/core/templates/pages/story-editor-page/chapter-editor/chapter-editor-tab.component.spec.ts#L44

Reply:
I've tried this, but it doesn't look like this solution as very scalable for test writers? I thought the entire point of using `UpgradedServices` is to avoid adding imports for Angular 8 services that "leak" into specs? For example, the following changes are indirectly forced to import these new dependencies:
```
core/templates/pages/exploration-editor-page/editor-navigation/editor-navbar-breadcrumb.component.spec.ts
core/templates/pages/exploration-editor-page/editor-navigation/editor-navigation.component.spec.ts
core/templates/pages/exploration-editor-page/editor-tab/exploration-editor-tab.component.spec.ts
core/templates/pages/exploration-editor-page/editor-tab/graph-directives/exploration-graph.component.spec.ts
core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts
core/templates/pages/exploration-editor-page/preview-tab/preview-tab.component.spec.ts
```

Reply:
Hm, I see. I think @srijanreddy98 would have some ideas. @srijanreddy98 few questions for you:
(1) When is it acceptable to use UpgradedServices in tests?
(2) If they need to be avoided, how do we make it easier for test writers to import required dependencies?

Thanks!

Reply:
As @brianrodri pointed out, yes the current solution doesn't scale well. I will come up with a proper solution for this, in the meanwhile, I would say it is valid to use ugs in specs where more than 5 angular services have to be manually listed.